tests/testthat/test-pandoc.R

```diff
@@ -0,0 +1,35 @@
+test_that("Correct syntax highlighting argument as requested", {
+  # helpers
+  highlight_style <- function(name = NULL) {
+    if (is.null(name))
+      "--no-highlight"
+    else
+      c("--highlight-style", name)
+  }
+  downlit  <- pandoc_variable_arg("highlight-downlit=1")
+  highlightjs <- pandoc_variable_arg("highlightjs=1")
+  a11y_theme <- highlight_style(pkg_file_highlight("a11y.theme"))
+  # check logic
+  expect_equal(pandoc_html_highlight_args("default", NULL), highlight_style())
+  expect_equal(pandoc_html_highlight_args("dummy.html", NULL), highlight_style())
+  expect_equal(pandoc_html_highlight_args("dummy.html", "default"),
```

Comment:
The wrapping makes it a little harder to see that you've covered all combinations of cases. It _might_ be worth making a local helper for `pandoc_html_highlight_args` that isn't quite so long. I also suspect re-ordering the arguments as I suggest above will make the logic more clear.